R/create_output_type_item.R

```diff
@@ -266,70 +315,73 @@ create_output_type_sample <- function(is_required, output_type_id_type, max_leng
   call <- rlang::current_call()
 
   schema <- download_tasks_schema(schema_version, branch)
+  pre_v4 <- hubUtils::version_lt("v4.0.0", schema_version = schema$`$id`)
 
   if (hubUtils::extract_schema_version(schema$`$id`) < "v3.0.0") {
     cli::cli_abort(
       "This function is only supported for schema versions {.val v3.0.0} and above."
     )
   }
   output_type_schema <- get_schema_output_type(schema, "sample")
-  output_type_id_schema <- purrr::pluck(
+  output_type_id_params_schema <- purrr::pluck(
     output_type_schema,
     "properties",
     "output_type_id_params",
     "properties"
   )
-
   type <- output_type_id_type # nolint: object_usage_linter
-  scalar_params <- c(
-    "is_required", "type", "min_samples_per_task",
-    "max_samples_per_task"
-  )
-  if (!is.null(max_length)) {
-    scalar_params <- c(scalar_params, "max_length")
+
```

Comment:
I'm now creating the objects below earlier in the function and passing them to `check_properties_*` fns to do the checking